lobsterpy/featurize/core.py

```diff
@@ -0,0 +1,829 @@
+# Copyright (c) lobsterpy development team
+# Distributed under the terms of a BSD 3-Clause "New" or "Revised" License
+
+"""
+This module defines classes to featurize Lobster data ready to be used for ML studies
+"""
+
+from __future__ import annotations
+import gzip
+import json
+import os
+import warnings
+from pathlib import Path
+from typing import List, Tuple
+from collections import namedtuple
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+from mendeleev import element
+from pymatgen.core.structure import Structure
+from pymatgen.io.lobster import Charge, Icohplist, MadelungEnergies
+from pymatgen.electronic_structure.cohp import CompleteCohp
+from pymatgen.electronic_structure.core import Spin
+from scipy.integrate import trapezoid
+from lobsterpy.cohp.analyze import Analysis
+
+warnings.filterwarnings("ignore")
+
+
+class FeaturizeLobsterpy:
+    """
+    class to featurize lobsterpy data
+
+    Args:
+        path_to_json: path to lobster lightweight json
+        bonds: "all_bonds" or "cation_anion_bonds"
+    Attributes:
+        get_df: returns a pandas dataframe with relevant icohp statistical data as columns from
+        lobsterpy automatic bonding analysis
+
+    """
+
+    def __init__(
+        self,
+        path_to_lobster_calc: str | None = None,
+        path_to_json: str | None = None,
+        bonds: str = "all_bonds",
```

Comment:
Strings are not consistent with the rest of the code. I think we use "all" and "cation-anion" elsewhere.